src/pages/PlaylistPage.js

```diff
@@ -1,40 +1,138 @@
+import * as actionCreators from '../action_creators';
+import {capitalize} from '../util';
+import {
+  Button,
+  Col,
+  Collapse,
+  Grid,
+  Row
+} from 'react-bootstrap';
 
+export const PlaylistPage = React.createClass({
+  getInitialState() {
+    return {
+      showMobileFilter: false,
+      showMobilePlaylists: false,
+      showMobileLevelNavigation: false
+    };
+  },
+  getCourse() {
+    const courses = this.props.courses;
+    const courseName = this.props.params.course.toLowerCase();
+    return courses[courseName] || {};
+  },
+  getLessons(course) {
+    const lessons = course.lessons;
+    if (lessons == null) return {};
+
+    // Get lessons sorted by level
+    return Object.keys(lessons).reduce((res, lessonId) => {
+      const lesson = lessons[lessonId];
+      const level = lesson.level;
 
-const PlaylistPage = React.createClass({
-  getLessonPath(path) {
-    const lessonPath = /^\.(.*)\.md/.exec(path); // get the '/course/lesson/file' part of the path
-    return lessonPath ? lessonPath[1] : null;
+      // Check if lesson matches filter and belongs to this course
+      if (level != null) {
+        if (res.hasOwnProperty(level)) res[level].push(lesson);
+        else res[level] = [lesson];
+      }
+
+      return res;
+    }, {});
   },
   render() {
-    // Get all ./*/*/*.md under lessonSrc, e.g. ./scratch/straffespark/straffespark.md
-    // Ignorerer alle ./*/*/README.md
-    const lessonContext = require.context('onlyFrontmatter!lessonSrc/', true,
-      /^\.\/[^\/]*\/[^\/]*\/(?!README\.md$)[^\/]*\.md/);
-    //console.log('PlaylistPage lessonContext keys:');
-    //console.log(lessonContext.keys());
-
-    const keys = lessonContext.keys().filter(path =>{
-      return path.startsWith('./' + this.props.params.course);
-    });
-    const courseNodes = keys.map((path, idx) => {
-      const lesson = lessonContext(path);
-      const fm = lesson.frontmatter;
-      const lessonPath = this.getLessonPath(path);
-      const lessonStr = fm.title + (fm.level ? ` (Level ${fm.level})` : '');
-      return (
-        <div key={idx}>
-          {lessonPath ? <NavLink to={lessonPath}>{lessonStr}</NavLink> : lessonStr }
-        </div>
-      );
-    });
-    //console.log(keys);
-    return (
-      <div>
-        PlaylistPage for {this.props.params.course}
-        {courseNodes}
+    const course = this.getCourse();
+    const lessons = this.getLessons(course);
+    const levels = Object.keys(lessons);
+    const lessonLists = Object.keys(lessons).map((level, idx) => (
+      <div key={idx} className='col-xs-12'>
+        <LessonList id={'level-' + level} level={level} lessons={lessons[level]}/>
       </div>
+    ));
+    const showLevelNavigationMobile = (course.lessons || []).length > 10 && levels.length > 1;
+    const showLevelNavigationDesktop = (course.lessons || []).length > 15 && levels.length > 1;
+    return (
+      <Grid fluid={true}>
+
+        {/*Header*/}
+        <Row>
```

Comment:
Jeg spør fordi de er så spesifikke til PlaylistPage, ja. Det virker fornuftig at pages skal være "skall", og innhold komme fra components. Det bare virker litt rart å skulle blande components som er generelle og laget for gjenbruk med components som er veldig spesifikke til èn side. Men hvis vi legger de under `components/PlaylistPage`, `components/FrontPage`, `components/LessonPage`... eller noe sånt, så blir det jo ikke noe problem.